src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1002,24 +997,16 @@ private RegexNode ScanReplacement()
                         _pos = parenPos - 1;       // jump to the start of the parentheses
                         _ignoreNextParen = true;    // but make sure we don't try to capture the insides
 
-                        int charsRight = _pattern.Length - _pos;
-                        if (charsRight >= 3 && _pattern[_pos + 1] == '?')
+                        if (_pos + 2 < _pattern.Length && _pattern[_pos + 1] == '?')
                         {
-                            char rightchar2 = _pattern[_pos + 2];
-
                             // disallow comments in the condition
-                            if (rightchar2 == '#')
+                            if (_pattern[_pos + 2] == '#')
```

Comment:
The removal of rightchar2 is reintroducing several bounds-checked string accesses that weren't there before.

Reply:
the jit appears to be [smart enough](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgAYBYAKDIEYabi6BOACgCJAeDcBB9jgSgDcjWnSQACYinEAxNmAAW2KAG0AuuID6AB2wYMMKADs04gJZGMW7RFz8aAbxHiX5y+MXLcAJTMBzBSsAXms9A2MAOgAZGCM/DAVxOGtbYWpXcwAzcXklKB9/QPEAPhCAZnEAMkrQ/UMjFR1bcQBqcToNIJCAcgB+bvt01ychjJdPKHEoQowJ0nEQnTD6xptcVvFSNTSaMdcAen3xABMzXGwAGwuIAHcPCHx8WIx1i3EEmHujU4wzCCNdnssjlpgFZnl5l1xN0AMQDQF7EZAsYJKC3cRGGB3ACiCDAMG0v3+bCECLGAF9nEDDiczpdrncjNgnscPNhCQBXWDiPxojnaHK9AA8EQixVF/Dc7wUn0g3zMRIBozGZmybFBgTmCx6AB1uvDlRkkcjXKj0ZicXiCYqSWlkZTqGSMqqchMCmCSiFpNUpjMtVDukLulUaks6sZVs02mUNABCHqx4M+sPhBpNdbRuM9IIG5HGk3StGMrHiXH4wl/Iy2p2uB0ZB0OkTMCRSWSkXLKdS1VOmCxWdODfNuKxu3wexa6cNGaKxeKJZLpu2uF0d/JjoqlcQVZOT1ORjPtTo9fqDI01lw007nK7oyCPZ6vIzS2X/H6V8/AtgplbpjZbbXQnCp5AkOyJmsWlrljapKGrWVJ7JedI3oyzIwKyYDshgXKfLyED8oKIpihKUofF8b7/B+K7fhGv5tP+AZ6rmIEfhk4EYiWZbWpW1awS4ja8Z+o4zJ64jeqGu4/msf5HtCQYht2klRluWbQom8nUWmUmZuI8bQjmwGIixpoKEW7GQVxxIwfaZINkAA)?

Reply:
> the jit appears to be smart enough

_pattern and _pos are fields, not parameters